assets/src/onboarding-wizard/pages/done/index.js

```diff
@@ -0,0 +1,194 @@
+
+	if ( savingOptions || savingDeveloperToolsOption || downloadingTheme || hasOptionsChanges ) {
+		return <Saving />;
+	}
+
+	return (
+		<div className="done">
+			<h1 className="done__heading">
+				{ __( 'Done', 'amp' ) }
+			</h1>
+			<div className="done__content done__content--primary">
+				<h2 className="done__icon-title">
+					<IconLaptopSearch />
+					{ __( 'Review', 'amp' ) }
+				</h2>
+				{ READER === themeSupport && downloadedTheme === readerTheme && (
+					<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_SUCCESS }>
+						{ __( 'Your Reader theme was automatically installed', 'amp' ) }
+					</AMPNotice>
+				) }
+				{ readerModeWasOverridden && (
+					<AMPNotice type={ NOTICE_TYPE_INFO } size={ NOTICE_SIZE_LARGE }>
+						{ __( 'Because you selected a Reader theme that is the same as your site\'s active theme, your site has automatically been switched to Transitional template mode.', 'amp' ) }
+					</AMPNotice>
+				) }
+				<p>
+					{ __( 'Your site is ready to bring great experiences to your users!', 'amp' ) }
+				</p>
+				{ STANDARD === themeSupport && (
+					<p>
+						{ __( 'In Standard mode there is a single AMP version of your site. Browse your site below to ensure it meets your expectations.', 'amp' ) }
+					</p>
+				) }
+				{ TRANSITIONAL === themeSupport && (
+					<>
+						<p>
+							{ __( 'In Transitional mode AMP and non-AMP versions of your site are served using your currently active theme.', 'amp' ) }
+						</p>
+						<p>
+							{ __( 'Browse your site below to ensure it meets your expectations, and toggle the AMP setting to compare both versions.', 'amp' ) }
+						</p>
+					</>
+				) }
+				{ READER === themeSupport && (
+					<>
+						<p>
+							{ __( 'In Reader mode AMP is served using your selected Reader theme, and pages for your non-AMP site are served using your primary theme. Browse your site below to ensure it meets your expectations, and toggle the AMP setting to compare both versions.', 'amp' ) }
+						</p>
+						<p>
+							{ __( 'As a last step, use the Customizer to tailor the Reader theme as needed.', 'amp' ) }
+						</p>
+					</>
+				) }
+				<div className="done__links-container">
+					<NavMenu
+						links={ previewLinks }
+						onClick={ ( e, link ) => {
+							e.preventDefault();
+							setActivePreviewLink( link );
+						} }
+					/>
+				</div>
+			</div>
+			<div className="done__preview-container">
+				{ READER === themeSupport && downloadingThemeError && (
+					<AMPNotice size={ NOTICE_SIZE_LARGE } type={ NOTICE_TYPE_INFO }>
+						{ __( 'There was an error downloading your Reader theme. As a result, your site is currently using the legacy reader theme. Please install your chosen theme manually.', 'amp' ) }
+					</AMPNotice>
+				) }
+				{ STANDARD !== themeSupport && (
+					<AMPSettingToggle
+						text={ __( 'AMP', 'amp' ) }
+						checked={ isPreviewingAMP }
+						onChange={ toggleIsPreviewingAMP }
+						compact={ true }
+					/>
+				) }
```

Comment:
I noticed that on non-AMP there is a gap appearing in the page where the admin bar normally is:


https://user-images.githubusercontent.com/134745/135026552-d3ffa06d-3bad-4a0b-8173-fb9156f77931.mov

The issue seems to to be that the conditional in the following code is not allowing the `script` to be added to non-AMP pages:

https://github.com/ampproject/amp-wp/blob/73cfc46fe212d7dbb052121a995d2cb15e156322/includes/amp-helper-functions.php#L183-L219

It seems what is needed to fix this is to make the selector more specific:

```diff
diff --git a/includes/amp-helper-functions.php b/includes/amp-helper-functions.php
index d7fbc6f0d..8b55fc218 100644
--- a/includes/amp-helper-functions.php
+++ b/includes/amp-helper-functions.php
@@ -202,7 +202,7 @@ function() {
 					/** @type {HTMLStyleElement} */
 					const style = document.createElement( 'style' );
 					style.setAttribute( 'type', 'text/css' );
-					style.appendChild( document.createTextNode( 'html { margin-top: 0 !important; } #wpadminbar { display: none !important; }' ) );
+					style.appendChild( document.createTextNode( 'html:not(#_) { margin-top: 0 !important; } #wpadminbar { display: none !important; }' ) );
 					document.head.appendChild( style );
 
 					document.addEventListener( 'DOMContentLoaded', function() {
```

Otherwise, it is getting overridden:

![image](https://user-images.githubusercontent.com/134745/135027102-c79fbc30-150f-47a1-bfef-e4d5889d6a42.png)

Specifically by this style:

```html
<style media="screen">
	html { margin-top: 32px !important; }
	* html body { margin-top: 32px !important; }
	@media screen and ( max-width: 782px ) {
		html { margin-top: 46px !important; }
		* html body { margin-top: 46px !important; }
	}
</style>
```

Either that, or more simply the logic could be run at `wp_head` with priority 20, to go above the priority 10 at which `_admin_bar_bump_cb()` is outputting the above styles:

```diff
diff --git a/includes/amp-helper-functions.php b/includes/amp-helper-functions.php
index d7fbc6f0d..8b0235e23 100644
--- a/includes/amp-helper-functions.php
+++ b/includes/amp-helper-functions.php
@@ -187,7 +187,7 @@ static function() {
 	 * by Preview component located in <assets/src/setup/pages/save/index.js>.
 	 */
 	add_action(
-		'wp_print_scripts',
+		'wp_head',
 		function() {
 			if ( ! amp_is_dev_mode() || ! is_admin_bar_showing() ) {
 				return;
@@ -215,7 +215,8 @@ function() {
 				} )();
 			</script>
 			<?php
-		}
+		},
+		20
 	);
 }
```

We'd just have to make sure it also runs at `amp_post_template_head` I suppose.

Reply:
I really like the simplicity of the first approach with bumping up the selector specificity. I've implemented it in 28b4583.